packages/vaex-ml/vaex/ml/incubator/river.py

```diff
@@ -0,0 +1,137 @@
+    model = traitlets.Any(default_value=None, allow_none=True, help='A River model which implements the `learn_many` method.').tag(**serialize_pickle)
+    features = traitlets.List(traitlets.Unicode(), help='List of features to use.')
+    target = traitlets.Unicode(allow_none=False, help='The name of the target column.')
+    batch_size = traitlets.Int(default_value=1_000_000, allow_none=False, help='Number of samples to be sent to the model in each batch.')
+    num_epochs = traitlets.Int(default_value=1, allow_none=False, help='Number of times each batch is sent to the model.')
+    shuffle = traitlets.Bool(default_value=False, allow_none=False, help='If True, shuffle the samples before sending them to the model.')
+    prediction_name = traitlets.Unicode(default_value='prediction', help='The name of the virtual column housing the predictions.')
+
+    def __call__(self, *args):
+        X = {feature: np.asarray(arg, np.float64) for arg, feature in zip(args, self.features)}
+        X = pd.DataFrame(X)
+        return self.model.predict_many(X).values
+
+
+    def predict(self, df):
+        '''Get an in memory numpy array with the predictions of the Model
+
+        :param df: A vaex DataFrame containing the input features
+        :returns: A in-memory numpy array containing the Model predictions
+        :rtype: numpy.array
+        '''
+        return self.model.predict_many(df.to_pandas_df(column_names=self.features)).values
+
+
+    def transform(self, df):
+        '''
+        Transform A DataFrame such that it contains the predictions of the Model in a form of a virtual column.
+
+        :param df: A vaex DataFrame
+
+        :return df: A vaex DataFrame
+        :rtype: DataFrame
+        '''
+        copy = df.copy()
+        lazy_function = copy.add_function('river_model_prediction_function', self, unique=True)
+        expression = lazy_function(*self.features)
+        copy.add_virtual_column(self.prediction_name, expression, unique=False)
+        return copy
+
+
+    def fit(self, df, progress=None):
+        '''Fit the RiverModel to the DataFrame.
+
+        :param df: A vaex DataFrame containig the features and target on which to train the model
+        :param progress: If True, display a progressbar which tracks the training progress.
+        '''
+
+        # Check whether the model is appropriate
+        assert hasattr(self.model, 'learn_many')
+
+        n_samples = len(df)
+
+        progressbar = vaex.utils.progressbars(progress)
+
+        # Portions of the DataFrame to evaluate
+        expressions = self.features + [self.target]
+
+        for epoch in range(self.num_epochs):
+            for i1, i2, df_tmp in df.to_pandas_df(column_names=expressions, chunk_size=self.batch_size):
+                progressbar((n_samples * epoch + i1) / (self.num_epochs * n_samples))
+                y_tmp = df_tmp.pop(self.target)
+
+                if self.shuffle:
+                    shuffle_index = np.arange(len(df_tmp))
```

Comment:
this does not shuffle, do you want np.random?

Reply:
there was a line missing - fixed -thanks!